bin/utils.js

```diff
@@ -233,12 +231,18 @@ exports.setupWSConnection = (conn, req, { docName = req.url.slice(1).split('?')[
         conn.ping()
       } catch (e) {
         closeConn(doc, conn)
+        clearInterval(pingInterval)
       }
     }
   }, pingTimeout)
+  conn.on('close', () => {
+    closeConn(doc, conn)
+    clearInterval(pingInterval)
+  })
   conn.on('pong', () => {
     pongReceived = true
   })
+  
   // send sync step 1
   const encoder = encoding.createEncoder()
```

Comment:
`encoder` is in scope of the pingInterval handler which seems a bit unnecessary. I'll merge this PR, scope the encoder variable, and then publish a new release with your fix @ellisonbg :)